agent/agent.go

```diff
@@ -21,151 +18,145 @@ import (
 
 // Controls is the one struct to rule them all :)
 type controls struct {
-	wg           sync.WaitGroup
-	conn         *websocket.Conn
-	tokenRequest *pb.Request
-	setup        *pb.Setup
+	wg          *sync.WaitGroup
+	conn        *websocket.Conn
+	containerID string
+	setup       *pb.Setup
+	mutex       *sync.Mutex
 }
 
-const defaultHeartBeat = 10
+var agent *controls
```

Comment:
Should all this variables be global? Pls recheck

Even if yes - replace it with 
var {
}